torch/csrc/api/include/torch/detail/TensorDataContainer.h

```diff
@@ -209,6 +211,7 @@ AT_FORALL_SCALAR_TYPES_AND2(Half, BFloat16, TENSOR)
     }
 
     if (is_scalar()) {
+      at::tracer::impl::NoTracerDispatchMode tracer_guard;
       at::AutoNonVariableTypeMode non_var_type_mode(true);
```

Comment:
Delete AutoNonVariableTypeMode. I infer that these are tracing only modifications because there is an un-guarded `to` call in the else branch below